cookbooks/cdo-varnish/libraries/http_cache.rb

```diff
@@ -0,0 +1,100 @@
+# HTTP Cache configuration.
+#
+# See cdo-varnish/README.md for more information on the configuration format.
+class HttpCache
+  STATIC_ASSETS = {
+    # For static-asset extensions, don't forward any cookies or headers.
+    path: %w(cur pdf png gif jpeg jpg ico mp3 swf css js).map{|ext| "*.#{ext}"},
+    headers: [],
+    cookies: 'none'
+  }
+
+# Host header is needed for Varnish to delegate to pegasus/dashboard depending on the host.
+# Accept-Language header is needed to cache language-specific pages.
+  ALL_HEADERS = %w(Host Accept-Language)
```

Comment:
Yeah, the header-stripping part of the cache-configuration is still a bit unfinished- I need to do another pass at the documentation to make it clear what exactly the `header` field is used for in the current implementation, and what's still TODO. (Currently, `header` is only passed to the CloudFront distribution config but does not yet affect the Varnish configuration.)

For reference, here is the list of common HTTP request headers that CloudFront will forward to the origin without any additional configuration necessary, which should cover most typical HTTP usages: [HTTP Request Headers and CloudFront Behavior](http://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/RequestAndResponseBehaviorCustomOrigin.html#request-custom-headers-behavior)

There is an option to pass through all headers, which also disables caching entirely for the path (the syntax for this is `[*]` in the config hash, as documented), which we could use for paths where we worry about potentially unsafe stripping of headers. CloudFront does not support passing-through all but a specified list of headers to the origin.

The 'proxy' field is also currently partially-implemented, in the reverse direction- it is implemented in Varnish and the current configuration preserves the existing behavior, but is not yet implemented in CloudFront. I will need to finish the CloudFront implementation before we would be able to remove Varnish from our stack.